src/test/scala/chiselTests/LiteralToTargetSpec.scala

```diff
@@ -27,4 +27,64 @@ class LiteralToTargetSpec extends AnyFreeSpec with Matchers {
       ChiselStage.elaborate(new Foo)
     } should have).message("Illegal component name: UInt<1>(\"h01\") (note: literals are illegal)")
   }
+
+  "Vectors which are indexed by literal values should successfully be converted to Reference Target" in {
```

Comment:
or maybe this is just another case of the next one and shouldn't be expected to work either. We probably just need a better error message for "you're trying to annotate the result of a vector lookup, if you want to do that assign it to a temporary"